shared/src/vyxal/Elements.scala

```diff
@@ -2386,10 +2586,129 @@ object Elements:
         ListHelpers.makeIterable(a).vzip(ListHelpers.makeIterable(b))
+    },
+    addDirect(
+      "£",
+      "Set Register",
+      List("set-register", "->register", "set-reg", "->reg"),
+      Some(1),
+      "a: any -> register = a",
+    ) { ctx ?=>
+      ctx.globals.register = ctx.pop()
+    },
+    addDirect(
+      "¥",
+      "Get Register",
+      List("get-register", "get-reg", "register", "<-register", "<-reg"),
+      None,
+      " -> push the value of the register",
+    ) { ctx ?=>
+      ctx.push(ctx.globals.register)
+    },
+    addDirect(
+      "←",
+      "Rotate Stack Left",
+      List("rotate-stack-left"),
+      None,
+      " -> rotate the entire stack left once",
+    ) { ctx ?=>
+      ctx.rotateLeft
+    },
+    addDirect(
+      "→",
+      "Rotate Stack Right",
+      List("rotate-stack-right"),
+      None,
+      " -> rotate the entire stack right once",
+    ) { ctx ?=>
+      ctx.rotateRight
+    },
+    addDirect(
+      "\\",
+      "Dump",
+      List("dump"),
+      Some(1),
+      "a: any -> dump all values on the stack",
+    ) { ctx ?=>
+      ListHelpers.makeIterable(ctx.pop()).foreach(v => ctx.push(v))
+    },
+    addPart(
+      Monad,
+      "†",
+      "Length of Consecutive Groups",
+      List("len-consecutive", "gvl", "gavel"),
+      false,
+      "a: any -> lengths of consecutive groups of a",
+    ) {
+      case a =>
+        val iterable = ListHelpers.makeIterable(a)
+        val groups = ListHelpers.groupConsecutive(iterable)
+        VList.from(groups.map(ListHelpers.makeIterable(_).length))
+    },
+    addPart(
+      Monad,
+      "Π",
+      "Product",
+      List("product", "prod"),
+      false,
+      "a: lst -> product of a",
+    ) {
+      case a: VList => ListHelpers.product(a)
+      case a: String => ListHelpers.product(ListHelpers.makeIterable(a))
+      case a: VNum => ListHelpers.product(ListHelpers.makeIterable(a))
     }
 
-    // Constants
+    // Input reading
     ,
+    addDirect(
+      "⁰",
+      "First Input",
+      List("first-input", "input-0"),
+      Some(0),
+      "The first input to the program",
+    ) { ctx ?=>
+      ctx.globals.inputs(0)
+    },
+    addDirect(
+      "¹",
+      "Second Input",
+      List("second-input", "input-1"),
+      Some(0),
+      "The second input to the program",
+    ) { ctx ?=>
+      ctx.globals.inputs(1)
+    },
+    addDirect(
+      "²",
+      "Third Input",
+      List("third-input", "input-2"),
+      Some(0),
+      "The third input to the program",
+    ) { ctx ?=>
+      ctx.globals.inputs(2)
+    },
+    addPart(
+      Monad,
+      "⌈",
+      "Ceiling",
+      List("ceiling", "ceil"),
+      true,
+      "a: num -> ceil(a)",
+    ) {
+      case a: VNum => VNum(spire.math.ceil(a.underlying.real))
+    },
+    addPart(
+      Monad,
+      "⌊",
+      "Floor",
+      List("floor"),
+      true,
+      "a: num -> floor(a)",
+    ) {
+      case a: VNum => VNum(spire.math.floor(a.underlying.real))
```

Comment:
Does a.floor not work?

Reply:
It was originally for consistency because there was no `VNum.ceil`